base/ClientSweepAutomation.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+//Client sweep automation does the following - 
+//1. Starts with 50 threads and increments by 50 for every iteration. Each iteration is run for 10 seconds.
+//2. A % difference is calculated between the TPS of the current and previous iteration.
+//3. When there is a % difference of <= 1% for 7 consecutive iterations, the sweep is stopped (indicating a plateau). You can change this number 7 using the const variable plateau.
+//4. As the sweep progresses, the client threads corresponding to peak TPS values are saved. When there is more than 2% difference with a previous peak, a new peak is determined.
+//5. The sweep can go up to 20 iterations if the optimal number of client threads is not deduced earlier. You can change this number 20 using the const client_sweep_max_iterations.
+
+class ClientSweepAutomation {
+    const client_sweep_max_iterations = 20;
+    const plateau = 7;
+    private $curr_tps;
+    private $prev_tps;
+    private $max;
+    private $failedReq;
+    private $diff_percent;
+    private $plateau_tracker;
+    private $clientsweep_results;
+    private $optimalClientThreads;
+
+    public function __construct() {
+      $this->curr_tps = 1;
+      $this->prev_tps = 0;
+      $this->max = 0;
+      $this->failedReq = 0;
+      $this->diff_percent = 0;
+      $this->plateau_tracker = 0;
+      $this->clientsweep_results = array();
+      $this->optimalClientThreads = 0;
+    }
+
+    public function GetOptimalThreads(PerfOptions $options, PerfTarget $target, $mode) {
+      $options->clientThreads = 0;
+      for ($iteration = 1; $iteration <= self::client_sweep_max_iterations; $iteration++) {
+        $options->clientThreads += 50;
+        $siege = PerfRunner::RunSiege($options, $target, $mode);
+        $this->ParseSiegeOutput($siege, $options, $iteration);
+        if (!($this->CheckIfContinue($options, $iteration))) {
+          break;
+        }       
+      }
+      $this->PrintClientSweepResults();
+      if ($this->optimalClientThreads == 0) {
+        trigger_error("Client sweep automation could not figure optimal client threads", E_USER_ERROR);
+      }
+      return $this->optimalClientThreads;
+    }
+
+    private function ParseSiegeOutput(Siege $siege, PerfOptions $options, $iteration) {
+      $siege_stats = $siege->collectStats();
+      $this->curr_tps = $siege_stats['Combined']['Siege RPS'];
+      $this->failedReq = $siege_stats['Combined']['Siege failed requests'];
+      if ($iteration != 1) {
+        $this->diff_percent = (((floor($this->curr_tps - $this->prev_tps))/$this->prev_tps)*100);
+        $this->diff_percent = intval($this->diff_percent * ($p = pow(10,2))) / $p;
```

Comment:
Is this just for formatting the number? If so, how about use number_format()?

Reply:
Yes it is just for formatting. I will change it and use number_format(). I hadn't come across this function before but it does seem more readable.